core/ArchiveProcessor/Loader.php

```diff
@@ -63,20 +64,37 @@ protected function mustProcessVisitCount($visits)
 
     public function prepareArchive($pluginName)
     {
-        $this->params->setRequestedPlugin($pluginName);
-
-        list($idArchive, $visits, $visitsConverted) = $this->loadExistingArchiveIdFromDb();
-        if (!empty($idArchive)) {
-            return $idArchive;
+        // temporarily set the idSite query parameter so archiving will end up using
+        // the correct site aware caches
+        $originalGetIdSite = isset($_GET['idSite']) ? $_GET['idSite'] : null;
+        $originalPostIdSite = isset($_POST['idSite']) ? $_POST['idSite'] : null;
+
+        try {
+            $_GET['idSite'] = $_POST['idSite'] = $this->params->getSite()->getId();
```

Comment:
@diosmosis I tested it and it worked 👍  Its good to merge from my point of view. I'm now a little bit concerned by this here but I don't think it is actually a problem. I was thinking of random cases like what if `$_GET[idSites]=3` is set, we then go in here and archive the child sites `1` and `2` where we then set for example `$_GET[idSites]=1`. I was wondering if this could cause a problem when eg in the archiver another API is called eg via `API\Request::processRequest()` and suddenly we might end up with some random result and hard to explain bugs. For example https://github.com/matomo-org/matomo/blob/3.2.0/plugins/API/API.php#L269-L278 here we accept `idSites` and `idSite` (because we want to deprecate `idSites` but not break BC). Or some random behaviour for example with the caches in `siteAware`  (but I checked the code and it is fine). 

I don't think it is an actual problem though and more a problem in theory. To be safe I now wonder if we should also temporarily set `GET/POST[idSites]=$site->getId()` but not sure if this could cause a bug, don't think so. Maybe it would be safe to unset temporarily any possible `GET/POST[idSites]` value? 
I leave it up to you, I'm not quite sure. I'm thinking I would maybe `unset(idsites)` temporarily to make sure we work in the context of that site.